### PR TITLE
Allow to use some system libraries

### PR DESCRIPTION
**Pull request by symphorien**
* lua
* libssh
* libcmark

### CODE REVIEW

**Comment by yurivict** [comment]
With these patches it fails to find packages:
```
-- Checking for module 'libcmark'
--   Package 'libcmark', required by 'virtual:world', not found
CMake Error at dep/cmark/CMakeLists.txt:8 (add_subdirectory):
  The source directory

    /usr/ports/devel/gitahead/work/gitahead-2.5.5/dep/cmark/cmark

  does not contain a CMakeLists.txt file.


CMake Error at dep/cmark/CMakeLists.txt:9 (target_include_directories):
  Cannot specify include directories for target "libcmark_static" which is
  not built by this project.


-- Checking for module 'lua>=5.3'
--   Package 'lua', required by 'virtual:world', not found
-- Checking for module 'libsecret-1'
--   Found libsecret-1, version 0.18.6
-- Checking for module 'gnome-keyring-1'
--   Found gnome-keyring-1, version 3.12.0
-- Configuring incomplete, errors occurred!
```

```pkg-config``` finds ```lua```:
```
 pkg-config --cflags lua-5.3
-I/usr/local/include/lua53 
```

```libsecret``` is also installed:
```
$ pkg info | grep libsecret
libsecret-0.18.6_1             Library to access the secret service API
```

**Comment by symphorien** [comment]
and what does `pkg-config --cflags lua` say?

**Comment by yurivict** [comment]
```
$  pkg-config --cflags lua
Package lua was not found in the pkg-config search path.
Perhaps you should add the directory containing `lua.pc'
to the PKG_CONFIG_PATH environment variable
Package 'lua', required by 'virtual:world', not found
```

**Comment by symphorien** [comment]
Ok I pushed an untested commit for the case where lua-5.3 exists but not lua. Can you test it?

**Comment by yurivict** [comment]
It still can't find Lua.

**Comment by symphorien** [comment]
I guess I can only check for lua-5.3 and not lua. Are lua version usually incompatible with one another in such a way that accepting 5.4 does not make sense ?

**Comment by yurivict** [comment]
> Are lua version usually incompatible with one another in such a way that accepting 5.4 does not make sense ?

This could be.

**Comment by symphorien** [comment]
Here is an attempt at using pkg_search_module instead of pkg_check_modules, can you test it?

**Comment by yurivict** [comment]
Now it fails:
```
CMake Error at dep/git/CMakeLists.txt:5 (add_executable):
  Cannot find source file:

    /usr/ports/devel/gitahead/work/gitahead-2.5.5/dep/git/git/contrib/credential/gnome-keyring/git-credential-gnome-keyring.c
```

I have git installed.

**Comment by symphorien** [comment]
`ln -s /usr/share/git dep/git/git` should do the trick. Since this directory might depend on the distribution, I don't know how to automate discovery with cmake.

Note that this PR only claims to unbundle libcmark, libssh and lua.

**Comment by symphorien** [comment]
Is there something I can change in this PR to push things forward?

**Comment by hackhaslam** [comment]
@symphorien I'm sorry that I'm not very responsive. I will try to review this soon. The biggest question mark for me is how do these changes affect the Windows and macOS builds (particularly with regards to pkgconfig which won't be available by default)? I can test that, but do you have any ideas?

**Comment by symphorien** [comment]
Thanks for your interest.

As far as I understand, if pkg-config is not available, cmake should fallback to the current behavior of using the vendored dependencies. For this reason I expect that this does not break builds on windows or darwin.

File: dep/lua/CMakeLists.txt [hunk 1]
```diff
@@ -41,3 +45,6 @@ add_library(lua
 target_include_directories(lua INTERFACE
   ${CMAKE_CURRENT_SOURCE_DIR}/${LUA_DIR}/src
 )
+
+set(LUA_LIBRARIES lua CACHE INTERNAL "lua implementation" FORCE)
```
Comment (hackhaslam) [inline dep/lua/CMakeLists.txt]: This variable is never used. Was it supposed to be used or is it extraneous?

File: dep/lua/CMakeLists.txt [hunk 2]
```diff
@@ -1,3 +1,7 @@
+IF (PKG_CONFIG_FOUND)
+  pkg_search_module(LUA lua-5.3 lua5.3 lua>=5.3)
+ENDIF()
+IF (NOT LUA_FOUND)
```
Comment (hackhaslam) [inline dep/lua/CMakeLists.txt]: Please indent this block.

**Comment by symphorien** [comment]
I have tried to adress your comments.

**Pull request merged**